Do not use WTFMove

Constellation opened the pull request:
#### d0c2f303b61df6a455265e461cf45c6529a11516
<pre>
Do not use WTFMove
<a href="https://bugs.webkit.org/show_bug.cgi?id=304539">https://bugs.webkit.org/show_bug.cgi?id=304539</a>
<a href="https://rdar.apple.com/166926521">rdar://166926521</a>

Reviewed by Anne van Kesteren and Mark Lam.

Remove remaining WTFMove use in WebKit. Also, replace some of std::move
in JSC with WTF::move.

* Introduction.md:
* Source/JavaScriptCore/lol/LOLJIT.cpp:
(JSC::LOL::LOLJIT::compileAndLinkWithoutFinalizing):
(JSC::LOL::LOLJIT::privateCompileMainPass):
(JSC::LOL::LOLJIT::privateCompileSlowCases):
* Source/WTF/wtf/InlineWeakPtr.h:
(WTF::=):
(WTF::InlineWeakPtrHashTraits::take):
* Source/WTF/wtf/StdLibExtras.h:
* Source/WebCore/Modules/webdatabase/SQLTransaction.cpp:
(WebCore::SQLTransaction::deliverSuccessCallback):
* Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm:
(WebCore::AudioVideoRendererAVFObjC::notifyTimeReachedAndStall):
* Source/bmalloc/bmalloc/SegmentedVector.h:
* Websites/webkit.org/code-style.md:

Canonical link: <a href="https://commits.webkit.org/304817@main">https://commits.webkit.org/304817@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e0b72549f073061982c8046a67ae48edcf32fd7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136553 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8912 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47839 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144276 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/89526 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1888fe14-130c-4d88-ab4e-ca16e504971d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138425 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9604 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8758 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104420 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/74984 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3c88ea60-a4c1-4b38-b8ce-4e0cf6b160de) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139498 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7012 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122350 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85255 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/fb8aed4b-a318-48b0-bdd9-9486f54de4e5) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6656 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4326 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4872 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/128514 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115966 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40544 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147034 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/135039 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8596 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41119 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112760 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8614 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7228 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113102 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6579 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118649 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/62620 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21060 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8644 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36703 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/167818 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8363 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72210 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43784 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8584 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8436 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/bmalloc/bmalloc/SegmentedVector.h:
```diff
@@ -153,7 +153,7 @@ class SegmentedVector final {
     {
         BASSERT(!isEmpty());
-        T result = WTFMove(last());
+        T result = std::move(last());
```
This is bmalloc. We should not use WTF.

annevk reviewed (approved):
This seems more complete due to the Websites/ change.

MenloDorian reviewed (approved):
r=me

Constellation committed:
Do not use WTFMove
https://bugs.webkit.org/show_bug.cgi?id=304539
rdar://166926521

Reviewed by Anne van Kesteren and Mark Lam.

Remove remaining WTFMove use in WebKit. Also, replace some of std::move
in JSC with WTF::move.

* Introduction.md:
* Source/JavaScriptCore/lol/LOLJIT.cpp:
(JSC::LOL::LOLJIT::compileAndLinkWithoutFinalizing):
(JSC::LOL::LOLJIT::privateCompileMainPass):
(JSC::LOL::LOLJIT::privateCompileSlowCases):
* Source/WTF/wtf/InlineWeakPtr.h:
(WTF::=):
(WTF::InlineWeakPtrHashTraits::take):
* Source/WTF/wtf/StdLibExtras.h:
* Source/WebCore/Modules/webdatabase/SQLTransaction.cpp:
(WebCore::SQLTransaction::deliverSuccessCallback):
* Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm:
(WebCore::AudioVideoRendererAVFObjC::notifyTimeReachedAndStall):
* Source/bmalloc/bmalloc/SegmentedVector.h:
* Websites/webkit.org/code-style.md:

Canonical link: https://commits.webkit.org/304817@main
